extensions/parquet/table/src/main/java/io/deephaven/parquet/table/pagestore/OffsetIndexBasedColumnChunkPageStore.java

```diff
@@ -129,13 +130,10 @@ public ChunkPage<ATTR> getPageContaining(@NotNull final FillContext fillContext,
         int pageNum;
         if (fixedPageSize == PAGE_SIZE_NOT_FIXED) {
             pageNum = findPageNumUsingOffsetIndex(offsetIndex, row);
-            if (pageNum < 0) {
-                pageNum = -2 - pageNum;
-            }
         } else {
             pageNum = (int) (row / fixedPageSize);
             if (pageNum >= numPages) {
-                // This can happen if the last page is of different size from rest of the pages.
+                // This can happen if the last page is larger than rest of the pages, which are all the same size.
```

Comment:
This is a weird condition to account for; do we regularly expect that all of the pages will be the same size, _except_ the last page is _larger_?

IMO, if all the page sizes are the same, except the last is larger, we should just use `PAGE_SIZE_NOT_FIXED`.

We could equivalently come up with a formulation to account for the equally strange condition "All of the pages are the same size, except the first which is (larger|smaller)." and make efficient math for that, but it's weird.

Reply:
I don't think its a common case.
The challenging part is that we cannot find the size of last page using offset index. Offset index only has the index of first row on the page. So we don't know how many rows there are on the last page without actually reading the page.  So keeping it like this doesn't need the number of rows in the last page.

Reply:
Added a comment explaining this